app/modules/ph/endpoints_ph.py

```diff
@@ -0,0 +1,112 @@
+import uuid
+
+from fastapi import Depends, File, HTTPException, UploadFile
+from fastapi.responses import FileResponse
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from app.core import models_core, standard_responses
+from app.core.groups.groups_type import GroupType
+from app.core.module import Module
+from app.dependencies import (
+    get_db,
+    get_request_id,
+    is_user_a_member,
+    is_user_a_member_of,
+)
+from app.modules.ph import cruds_ph, models_ph, schemas_ph
+from app.utils.tools import get_file_from_data, save_file_as_data
+
+module = Module(
+    root="ph",
+    tag="ph",
+    default_allowed_groups_ids=[GroupType.student],
+)
+
+
+@module.router.get(
+    "/ph/{paper_id}/pdf",
+    response_class=FileResponse,
+    status_code=200,
+)
+async def get_paper_pdf(
+    paper_id: str,
+    user: models_core.CoreUser = Depends(is_user_a_member),
+):
+    return get_file_from_data(
+        default_asset="assets/pdf/default_PDF.pdf",
+        directory="ph",
+        filename=str(paper_id),
+    )
+
+
+@module.router.get(
+    "/ph/",
+    response_model=list[schemas_ph.PaperComplete],
+    status_code=200,
+)
+async def get_papers(
+    db: AsyncSession = Depends(get_db),
+    user: models_core.CoreUser = Depends(is_user_a_member),
+):
```

Comment:
Could you add a docstring explaining the results are sorted by date?

Reply:
In the crud or in the endpoint ?

Reply:
I would suggest in both the crud and the endpoint